includes/tracks/events/class-wc-products-tracking.php

```diff
@@ -40,6 +43,55 @@ public function track_product_updated( $product_id, $post ) {
 		WC_Tracks::record_event( 'product_edit', $properties );
 	}
 
+	/**
+	 * Track the Update button being clicked on the client side.
+	 * This is needed because `track_product_updated` (using the `edit_post`
+	 * hook) is called in response to a number of other triggers.
+	 *
+	 * @param WP_Post $post The post, not used.
+	 */
+	public function track_product_updated_client_side( $post ) {
+		wc_enqueue_js(
+			"
+			if ( $( 'h1.wp-heading-inline' ).text().trim() === 'Edit product') {
+				const initialStockValue = $( '#_stock' ).val();
+				let hasRecordedEvent = false;
+
+				$( '#publish' ).click( function() {
+					if ( hasRecordedEvent ) {
+						return;
+					}
+
+					const currentStockValue = $( '#_stock' ).val();
+					const properties = {
+						product_type:			$( '#product-type' ).val(),
+						is_virtual:				$( '#_virtual' ).is( ':checked' ) ? 'Y' : 'N',
+						is_downloadable:		$( '#_downloadable' ).is( ':checked' ) ? 'Y' : 'N',
+						manage_stock:			$( '#_manage_stock' ).is( ':checked' ) ? 'Y' : 'N',
+						stock_quantity_update:	( initialStockValue != currentStockValue ) ? 'Y' : 'N',
+					};
+
+					window.wcTracks.recordEvent( 'product_update', properties );
+					hasRecordedEvent = true;
+				} );
+			}
+			"
+		);
+	}
+
+	/**
+	 * Send a Tracks event when a product's stock level is adjusted.
+	 *
+	 * @param WC_Product $product Product.
+	 */
+	public function track_product_stock_level_set( $product ) {
+		$properties = array(
+			'product_id' => $product->get_id(),
```

Comment:
Similar to the order id, I don't think this property is useful in the context of Tracks.

Reply:
Actually this whole track isn't needed any more as stock level changes are now captured as requested in the JS handler above. I've removed this.